src/activities/ScoringEntity.js

```diff
@@ -0,0 +1,54 @@
+import { Entity } from '../es6/Entity';
+import { Actions } from '../hypermedia-constants';
+import { performSirenAction } from '../es6/SirenAction.js';
+
+/**
+ * ScoringEntity class representation of an activity's scoreOutOf subentity
+ */
+export class ScoringEntity extends Entity {
+	/**
+	 * @returns {string} Activitiy's scoreOutOf
+	 */
+	scoreOutOf() {
+		return this._entity && this._entity.properties && this._entity.properties.scoreOutOf;
+	}
+
+	/**
+	 * @returns {string} Activity's grade maxPoints, if there is a grade association
+	 */
+	gradeMaxPoints() {
+		return this._entity && this._entity.properties && this._entity.properties.gradeMaxPoints;
```

Comment:
With short circuit evaluation in JS, could this return a boolean instead a string when the earlier expressions are false?

Reply:
I believe if any value is `undefined` then the whole thing will evaluate to `undefined`, and not a boolean value.